### PR TITLE
Set Win Condition to All Non-Mine Spaces Opened

### LINKED ISSUE #1: You can still win the game if wrong flags have been placed
If you place more flags than there are mines, its still possible to win the game.
Here an extreme example of it:
https://asciinema.org/a/bqGkvtZjIztnQmpd3cLSgAv5D

### PR DESCRIPTION
**Pull request by LRFLEW**
I have a feeling that all the PR's I'm going to make on this project are going to contain the phrase "In the Windows version". It's mostly because it was a really well designed implementation, and understanding *why* they made certain choices is useful. I'm really enjoying this project, and have been playing it since I found it.

Anyways, this fixes #1 by changing the tracked variable for winning, from being the number of flagged mines to the number of opened non-mine spaces. You seemed a bit confused when I mentioned this in the issue, but the flags actually don't have anything to do with the win condition in the Windows version. The flags were there to visually help the user, and not a required part of playing the game. You can see that by looking at [speedruns of the game](https://youtu.be/HDUVKUnr0I8?t=5s), where they often don't place flags when the mine locations are more obvious. This PR allows the game to be won without placing any flags, like the Windows version does.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -68,26 +69,24 @@ const Minesweeper = {
...
 
 
 	, abort: function () {
-		this.done = this.aborted = true
-		for (let y = 0; y < this.mines.length; y++) {
-			for (let x = 0; x < this.mines[0].length; x++) {
-				this.opened[x][y] = true
-			}
-		}
+		this.aborted = true
```
Comment (derhuerst) [inline index.js]: Put in `this.done` again.

File: index.js [hunk 2]
```diff
@@ -96,7 +95,6 @@ const Minesweeper = {
 
 	, submit: function () {
 		this.done = true
-		this.aborted = false
```
Comment (derhuerst) [inline index.js]: The reason for this that the code did not depend on the default values of `this.done` and `this.aborted`. Please put it back in.

File: index.js [hunk 3]
```diff
@@ -172,19 +171,20 @@ const Minesweeper = {
...
-				else if (isMine)      cell = isSelected ? '💥' : '💣'
-				else if (count === 1) cell = chalk.blue(count)
-				else if (count === 2) cell = chalk.green(count)
-				else if (count === 3) cell = chalk.yellow(count)
-				else if (count === 4) cell = chalk.red(count)
-				else if (count > 4)   cell = chalk.bold.red(count)
-				else                  cell = ' '
-
-				out += isSelected ? chalk.bgWhite.bold(cell) : cell
+				var cell
```
Comment (derhuerst) [inline index.js]: `let`.

**Comment by derhuerst** [comment]
Clearly you've had a lot more contact with MS Minesweeper than me, and therefore know a few details I didn't. Also, I didn't bother to manually check if my assumptions were true, so thanks for this PR.

Code-wise, the PR looks mostly good. I just have a few minor remarks.

File: index.js (hunk 1 above)
Comment (LRFLEW) [reply index.js]: Do you care if I keep `this.done` false here, or do you want it to be true like before? Setting it to true will require changing the render function again.

**Comment by LRFLEW** [comment]
@derhuerst any update on this? I made the changes requested, and asked for clarification on one of them.

File: index.js (hunk 1 above)
Comment (derhuerst) [reply index.js]: Keep it to `true` to keep it compatible with my other prompts. This is not documented anywhere, but I'd like `prompt-skeleton` to rely on it in the future.

**Comment by LRFLEW** [comment]
This should be ready now. Let me know if you have any more notes.

**Comment by derhuerst** [comment]
Apparently there are a few things to get wrong with MS Minesweeper. What do you think about putting all game logic into a separate lib-only npm package?

**Pull request merged**